### PR TITLE
CNS-812: verify TX arguments

### PR DESCRIPTION
**Pull request by oren-lava**

### CODE REVIEW

File: x/pairing/client/cli/tx_stake_provider.go [hunk 1]
```diff
@@ -188,6 +190,12 @@ func CmdBulkStakeProvider() *cobra.Command {
 						continue
 					}
 
+					q := spectypes.NewQueryClient(clientCtx)
```
Comment (Yaroms) [inline x/pairing/client/cli/tx_stake_provider.go:193]: rule of thumb is to not create a query when you do a transaction.
let it fail on the node side

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 2]
```diff
@@ -68,6 +84,14 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 			)
 		}
 
+		_, found := k.specKeeper.GetSpec(ctx, relay.SpecId)
```
Comment (Yaroms) [inline x/pairing/keeper/msg_server_relay_payment.go:87]: remove, this is not needed, happens in relay payment

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 3]
```diff
@@ -20,6 +21,21 @@ type BadgeData struct {
...
+	if len(msg.LatestBlockReports) > len(msg.Relays) {
+		return nil, utils.LavaFormatError("RelayPayment_invalid_latest_block_reports", fmt.Errorf("invalid latest block reports"),
+			utils.LogAttr("latestBlockReports", msg.LatestBlockReports),
+			utils.LogAttr("len(latestBlockReports)", len(msg.LatestBlockReports)),
+			utils.LogAttr("relays", msg.Relays),
+			utils.LogAttr("len(relays)", len(msg.Relays)),
+		)
+	}
+
+	if !commontypes.ValidateString(msg.GetDescriptionString(), commontypes.DESCRIPTION_RESTRICTIONS, nil) {
```
Comment (oren-lava) [inline x/pairing/keeper/msg_server_relay_payment.go:33]: this will fail if the description is nil, we don't want that

Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/b923474cd40ba656fc06008c7eb36a84a6652409

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 2 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/b923474cd40ba656fc06008c7eb36a84a6652409

**Review by Yaroms — APPROVED** [review]

**Review by Yaroms — APPROVED** [review]

**Pull request merged**